src/Verifier/SAW/Cryptol.hs

```diff
@@ -539,115 +543,136 @@ proveProp sc env prop =
-    "#"             -> scGlobalDef sc "Cryptol.ecCat"         -- {a,b,d} (fin a) => [a] d -> [b] d -> [a + b] d
-    "splitAt"       -> scGlobalDef sc "Cryptol.ecSplitAt"     -- {a,b,c} (fin a) => [a+b] c -> ([a]c,[b]c)
-    "join"          -> scGlobalDef sc "Cryptol.ecJoin"        -- {a,b,c} (fin b) => [a][b]c -> [a * b]c
-    "split"         -> scGlobalDef sc "Cryptol.ecSplit"       -- {a,b,c} (fin b) => [a * b] c -> [a][b] c
-    "reverse"       -> scGlobalDef sc "Cryptol.ecReverse"     -- {a,b} (fin a) => [a] b -> [a] b
-    "transpose"     -> scGlobalDef sc "Cryptol.ecTranspose"   -- {a,b,c} [a][b]c -> [b][a]c
-    "@"             -> scGlobalDef sc "Cryptol.ecAt"          -- {n, a, ix} (Integral ix) => [n]a -> ix -> a
-    "!"             -> scGlobalDef sc "Cryptol.ecAtBack"      -- {n, a, ix} (fin n, Integral ix) => [n]a -> ix -> a
-    "update"        -> scGlobalDef sc "Cryptol.ecUpdate"      -- {n, a, ix} (Integral ix) => [n]a -> ix -> a -> [n]a
-    "updateEnd"     -> scGlobalDef sc "Cryptol.ecUpdateEnd"   -- {n, a, ix} (fin n, Integral ix) => [n]a -> ix -> a -> [n]a
-
-    -- Enumerations
-    "fromTo"        -> scGlobalDef sc "Cryptol.ecFromTo"
-                               -- fromTo : {first, last, bits, a}
-                               --           ( fin last, fin bits, last >== first,
-                               --             Literal first a, Literal last a)
-                               --        => [1 + (last - first)]a
-    "fromThenTo"    -> scGlobalDef sc "Cryptol.ecFromThenTo"
-                               -- fromThenTo : {first, next, last, a, len}
-                               --              ( fin first, fin next, fin last
-                               --              , Literal first a, Literal next a, Literal last a
-                               --              , first != next
-                               --              , lengthFromThenTo first next last == len) => [len]a
-
-    "error"         -> scGlobalDef sc "Cryptol.ecError"       -- {at,len} (fin len) => [len][8] -> at -- Run-time error
-    "random"        -> scGlobalDef sc "Cryptol.ecRandom"      -- {a} => [32] -> a -- Random values
-    "trace"         -> scGlobalDef sc "Cryptol.ecTrace"       -- {n,a,b} [n][8] -> a -> b -> b
-    "Array::arrayConstant" -> scGlobalDef sc "Cryptol.ecArrayConstant" -- {a,b} b -> Array a b
-    "Array::arrayLookup"   -> scGlobalDef sc "Cryptol.ecArrayLookup" -- {a,b} Array a b -> a -> b
-    "Array::arrayUpdate"   -> scGlobalDef sc "Cryptol.ecArrayUpdate" -- {a,b} Array a b -> a -> b -> Array a b
-
-    _ -> panic "Unknown Cryptol primitive name" [C.unpackIdent nm]
-
-importPrimitive _ nm =
-  panic "Improper Cryptol primitive name" [show nm]
+importPrimitive sc n
+  | Just nm <- C.asPrim n, Just term <- Map.lookup nm (prelPrims <> floatPrims) = term sc
+  | Just nm <- C.asPrim n = panic "Unknown Cryptol primitive name" [show nm]
+  | otherwise = panic "Improper Cryptol primitive name" [show n]
+
```

Comment:
This is just stylistic, but I'd
  1. add separate top-level declaration for the map of translations (e.g. `allPrims = prelPrims <> floatPrims <> arrayPrim`)
  2. I'd reduce the guards to 2 as that matches the logic more directly: either the name is a primitve or not.  In the case for the primitive I'd use a `case` to lookup in `allPrims` and `panic` if the primitive is missing.